pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -272,17 +276,25 @@ func (r *ReconcileNode) deleteCrashCollector(deployment appsv1.Deployment) error
 }
 
 func (r *ReconcileNode) reconcileCrashRetention(namespace string, cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion) error {
+
+	k8sVersion, err := k8sutil.GetK8SVersion(r.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch get version")
+	}
 	if cephCluster.Spec.CrashCollector.DaysToRetain == 0 {
 		logger.Debug("deleting cronjob if it exists...")
-		cronJob := &v1beta1.CronJob{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      prunerName,
-				Namespace: namespace,
-			},
+
+		var errors error
+		// minimum k8s version required for v1 cronJob is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0',  else apply v1beta1 cronJob.
+		if k8sVersion.AtLeast(version.MustParseSemantic(minVersionForCronV1)) {
+			cronJob := createV1CronJob(namespace)
+			errors = r.client.Delete(context.TODO(), cronJob)
```

Comment:
We can move the `delete` operation out of the `if` block. 
``` 
		var cronJob client.Object
		// minimum k8s version required for v1 cronJob is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0',  else apply v1beta1 cronJob.
		if k8sVersion.AtLeast(version.MustParseSemantic(minVersionForCronV1)) {
			cronJob = createV1CronJob(namespace)
		} else {
			cronJob = createV1Beta1CronJob(namespace)
		}

		err := r.client.Delete(context.TODO(), cronJob)
```

Reply:
moved out.